server/datastore/mysql/hosts.go

```diff
@@ -2141,7 +2141,36 @@ func (ds *Datastore) getOrInsertMunkiIssues(ctx context.Context, errors, warning
 	return msgToID, nil
 }
 
-func (ds *Datastore) SetOrUpdateMDMData(ctx context.Context, hostID uint, isServer, enrolled bool, serverURL string, installedFromDep bool, name string) error {
+func (ds *Datastore) SetOrUpdateMDMData(
+	ctx context.Context,
+	hostID uint,
+	isServer, enrolled bool,
+	serverURL string,
+	installedFromDep bool,
+	name string,
+) error {
+	// MDM queries return an empty server URL when the host is not enrolled to an MDM.
+	if serverURL == "" {
+		// We use the reader even if it might miss some not replicated rows,
+		// because it will eventually catch up on subsequent ingestions and
+		// the entry will be deleted.
+		var id uint
+		switch err := sqlx.GetContext(ctx, ds.reader, &id,
+			`SELECT host_id FROM host_mdm WHERE host_id = ?`, hostID,
```

Comment:
Do we need to do a select first? Why can't we just delete it?

Reply:
To avoid using the `ds.writer` on every host `/distributed/write`.